colossalai/tensor/utils.py

```diff
@@ -164,3 +165,17 @@ def convert_parameter(module: torch.nn.Module, param_name: str):
 
     # Now we can set the attribute appropriately.
     setattr(module, param_name, st)
+
+
+def convert_dim_partition_dict(dim_size: int, dim_partition_dict: Dict[int, List[int]]) -> Dict[int, List[int]]:
+    '''
+    This method is used to convert the negative dim value to positive.
+    '''
+    dims_to_convert = []
+    for dim, mesh_list in dim_partition_dict.items():
+        if dim < 0:
+            dims_to_convert.append(dim)
+    for dim in dims_to_convert:
+        dim_partition_dict.pop(dim)
+        dim_partition_dict[dim_size + dim] = mesh_list
```

Comment:
mesh_list is the last value in dict `dim_partition_dict`.
Do you mean that all the mesh_lists are the same?

Reply:
'dim_partition_dict' is a dictionary type datastructure whose key is dimension to be sharded and value is the mesh list decribing the sharding axises. Actually, negative dim value is okay if there is no further action on this datastructure. However, transposes or even squeeze may happen during logical shape recovering to physical shape, which means negative dim value may potentially introduce an invisible bug. 
For example, linear weight dim partition dict is {0: [0], -1:[1]}, if transpose logic for linear weight convert dim 0 into 1 and dim 1 into 0, the dim partition dict finally looks like {1: [0], -1: [1]}.